src/common/auth.test.js

```diff
@@ -0,0 +1,29 @@
+import { describe, expect, it } from "vitest";
+import { validateSession } from "./auth.js";
+
+describe("auth", () => {
+  it("should validate session", () => {
+    const session = {
+      profile: {
+        displayName: "Joseph Bloggs",
+      },
+    };
+    expect(validateSession({}, session)).toEqual({
+      isValid: true,
+      credentials: session,
+    });
+  });
+
+  it("should return invalid if no session", () => {
+    expect(validateSession({}, null)).toEqual({
+      isValid: false,
+    });
+  });
+
+  it("should return invalid if no session.profile", () => {
+    const session = {};
+    expect(validateSession({}, session)).toEqual({
+      isValid: false,
+    });
+  });
+});
```

Comment:
Perhaps a better test would be to register with `hapi.server`?

It will cover real plugin behaviour (validation on registration etc) and be a single place where the logic is tested irrespective of the other routes existing or not. Atm, `auth` is best covered in tests for `getSecretRoute` which we'll delete.